### DISCUSSION TITLE
manual promotion with an expiry and rollout aborts if not done within expiry

### DISCUSSION THREAD

**Discussion by gaurav517** [argoproj/argo-rollouts#1087] [Q&A] [answered]
In canary strategy, is there a way to specify the manual promotion with an expiry and rollout gets aborted if manual promotion is not done within expiry.

For instance, we want users to promote rollout manually but if it its not promoted manually in one hour, rollout gets aborted.

Thanks.

**Comment by kostis-codefresh** [comment]
That would be an interesting feature request. Maybe you can open a Github issue for this?

**Comment by jessesuen** [comment, marked as answer]
It's not currently possible using any built-in features of Rollouts. However, you can approximate the behavior by doing the following

Have an inline AnalysisTemplate which is just a Kubernetes job that simply runs the command:

```shell
sleep 1h && exit 1
```

After one hour, the job will exit unsuccessfully, causing the rollout to abort. But if the rollout was manually promoted before the job completed, the Job would be terminated (deleted) and Rollout would be considered successful.
